File: packages/services/Car/car-lib/src/com/android/car/internal/common/CommonConstants.java
/*
 * Copyright (C) 2020 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.android.car.internal.common;

import static com.android.car.internal.ExcludeFromCodeCoverageGeneratedReport.BOILERPLATE_CODE;

import android.annotation.IntDef;

import com.android.car.internal.ExcludeFromCodeCoverageGeneratedReport;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Provides common constants for Car library, Car Service, and System Server.
 *
 * @hide
 */
public final class CommonConstants {

    @ExcludeFromCodeCoverageGeneratedReport(reason = BOILERPLATE_CODE)
    private CommonConstants() {
        throw new UnsupportedOperationException("contains only static constants");
    }

    // CarUserManagerConstants

    public static final int USER_LIFECYCLE_EVENT_TYPE_STARTING = 1;
    public static final int USER_LIFECYCLE_EVENT_TYPE_SWITCHING = 2;
    public static final int USER_LIFECYCLE_EVENT_TYPE_UNLOCKING = 3;
    public static final int USER_LIFECYCLE_EVENT_TYPE_UNLOCKED = 4;
    public static final int USER_LIFECYCLE_EVENT_TYPE_STOPPING = 5;
    public static final int USER_LIFECYCLE_EVENT_TYPE_STOPPED = 6;

    // CarService Constants
    public static final String CAR_SERVICE_INTERFACE = "android.car.ICar";

    @IntDef(prefix = { "USER_LIFECYCLE_EVENT_TYPE_" }, value = {
            USER_LIFECYCLE_EVENT_TYPE_STARTING,
            USER_LIFECYCLE_EVENT_TYPE_SWITCHING,
            USER_LIFECYCLE_EVENT_TYPE_UNLOCKING,
            USER_LIFECYCLE_EVENT_TYPE_UNLOCKED,
            USER_LIFECYCLE_EVENT_TYPE_STOPPING,
            USER_LIFECYCLE_EVENT_TYPE_STOPPED,
    })
    @Retention(RetentionPolicy.SOURCE)
    public @interface UserLifecycleEventType{}
}

File: platform_testing/libraries/flicker/test/src/com/android/server/wm/flicker/CommonConstants.kt
/*
 * Copyright (C) 2021 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@file:JvmName("CommonConstants")
package com.android.server.wm.flicker

import com.android.server.wm.traces.common.FlickerComponentName

val CHROME_COMPONENT = FlickerComponentName("com.android.chrome",
        "org.chromium.chrome.browser.firstrun.FirstRunActivity")
val CHROME_SPLASH_SCREEN_COMPONENT = FlickerComponentName("", "Splash Screen com.android.chrome")
val DOCKER_STACK_DIVIDER_COMPONENT = FlickerComponentName("", "DockedStackDivider")
val IMAGINARY_COMPONENT = FlickerComponentName("", "ImaginaryWindow")
val IME_ACTIVITY_COMPONENT = FlickerComponentName("com.android.server.wm.flicker.testapp",
        "com.android.server.wm.flicker.testapp.ImeActivity")
val LAUNCHER_COMPONENT = FlickerComponentName("com.google.android.apps.nexuslauncher",
        "com.google.android.apps.nexuslauncher.NexusLauncherActivity")
val PIP_DISMISS_COMPONENT = FlickerComponentName("", "pip-dismiss-overlay")

val SIMPLE_APP_COMPONENT = FlickerComponentName("com.android.server.wm.flicker.testapp",
        "com.android.server.wm.flicker.testapp.SimpleActivity")
private const val SHELL_PKG_NAME = "com.android.wm.shell.flicker.testapp"
val SHELL_SPLIT_SCREEN_PRIMARY_COMPONENT = FlickerComponentName(SHELL_PKG_NAME,
        "$SHELL_PKG_NAME.SplitScreenActivity")
val SHELL_SPLIT_SCREEN_SECONDARY_COMPONENT = FlickerComponentName(SHELL_PKG_NAME,
        "$SHELL_PKG_NAME.SplitScreenSecondaryActivity")

val SCREEN_DECOR_COMPONENT = FlickerComponentName("", "ScreenDecorOverlay")
val WALLPAPER_COMPONENT = FlickerComponentName(
        "", "com.breel.wallpapers18.soundviz.wallpaper.variations.SoundVizWallpaperV2")
